application.py

```diff
@@ -204,16 +204,30 @@ def handleUSECluster3():
         raise BeagleError(errors.TOO_FEW_QUESTIONS) #Communicate we don't support
     
     embeddings, data_used_for_demo, q_ids_list = get_data_embeddings(data)
```

Comment:
Hi, so as of now, I am not making this change. This is because of the following reasons:

1. This is a pretty big change since it would require re-factoring a lot of functions, across 2 files
2. I believe the current method is a bit more tedious, but is also very well documented and explained throughout the codebase :D 
3. The clustering is working well now, and all the functions are well defined and intuitive to understand. It would take some time for a new person to go through the codebase and understand the process, but I believe even after this change, someone new will have to still go through the code to understand the clustering, since there are a lot of processes involved :D 
4. I don't want to refactor everything now, since it can end up breaking more code than helping, and since we have a time constraint right now. However, in the future, we can consider refactoring it to this way :D 
5.  Thank you for bringing this up! Ideally, we should have decided on this format when we were creating the code in the beginning itself, to avoid so much refactoring, but this teaches us the importance of planning ahead and creating a pipeline before jumping into the implementation :D 

Cheers!